src/fieldml_util_routines.f90

```diff
@@ -207,8 +207,6 @@ SUBROUTINE FIELDML_IO_FINALISE( FIELDML_INFO, ERR, ERROR, * )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_CONNECTIVITY_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_LAYOUT_HANDLES, ERR, ERROR, *999 )
       DEALLOCATE(FIELDML_INFO)
```

Comment:
The internal FieldML Info is a pointer member of the external object. A call to public cmfe_FieldMLIO_Initialise only NULLIFYs the internal pointer; this appears to be consistent with other types.

The internal pointers themselves are allocated in the respective public CreateStart functions which are separate for Input and Output and an internal flag is set stating this 'direction'. The allocated internal pointer is passed as a non-pointer INTENT(INOUT) dummy argument to the internal initialise which fills its contents. This is an example of where it is allocated (in public cmfe_FieldML_OutputCreateObjVS):

`  ALLOCATE( fieldml%fieldmlInfo, stat=err )`
`  CALL FIELDML_OUTPUT_INITIALISE_INFO( mesh%mesh, location, baseName, connectivityFormat, fieldml%fieldmlInfo, &
      & err, error, *999 )`

The use of INTENT(INOUT) for a non-pointer dummy argument is probably fine for receiving already allocated pointers. However, it appears these allocations are not checked for success.

While in general test code submissions such as this should not fix internal issues that they do not test, if you agree that it is an improvement in style I'll move the allocations inside the internal CreateStart calls and test them, and obviously pass pointers as far as where allocations/deallocations occur. In all other places a pointer that's already proven as ASSOCIATED should be passed as an INTENT(INOUT) dummy argument so no further checks are needed.

Pah it doesn't look like the internal pointer is checked for being associated in most other API functions... can see if there's a tidy fix for this. I can certainly make a test which fails by crashing in this case.